ui/components/TopMenu/TopMenuProfileButton.tsx

```diff
@@ -1,31 +1,78 @@
 import { selectCurrentAccountTotal } from "@tallyho/tally-background/redux-slices/selectors"
-import React, { ReactElement } from "react"
+import { setSnackbarMessage } from "@tallyho/tally-background/redux-slices/ui"
+import React, { ReactElement, useState, useRef } from "react"
+import { useDispatch } from "react-redux"
 import { useBackgroundSelector } from "../../hooks"
 import SharedCurrentAccountInformation from "../Shared/SharedCurrentAccountInformation"
+import TopMenuProfileTooltip from "./TopMenuProfileTooltip"
+
+const TOOLTIP_DELAY = 500
 
 export default function TopMenuProfileButton(props: {
   onClick?: () => void
 }): ReactElement {
-  const { shortenedAddress, name, avatarURL } =
+  const dispatch = useDispatch()
+  const { shortenedAddress, name, avatarURL, address } =
     useBackgroundSelector(selectCurrentAccountTotal) ?? {}
 
+  const [shouldDisplayTooltip, setShouldDisplayTooltip] = useState(false)
+  const timerRef = useRef<number | undefined>(undefined)
+
   const { onClick } = props
 
+  const showTooltip = () => {
+    timerRef.current = window.setTimeout(
+      () => setShouldDisplayTooltip(true),
+      TOOLTIP_DELAY
+    )
+  }
+  const hideTooltip = () => {
+    clearTimeout(timerRef.current ?? 0)
+    setShouldDisplayTooltip(false)
+  }
+  const handleClick = () => {
+    hideTooltip()
+    onClick?.()
+  }
+
+  const copyAddress = () => {
+    if (address) {
+      navigator.clipboard.writeText(address)
+      hideTooltip()
+      dispatch(setSnackbarMessage("Address copied to clipboard"))
+    }
+  }
+
```

Comment:
There is a tiny "memory leak" here (not really relevant, but good to know).
```suggestion

  useEffect(() => () => {
    hideTooltip() // cleanup
  })
```
to make sure the timeout is never triggered when the component is not mounted.